code/__byond_version_compat.dm

```diff
@@ -61,3 +61,15 @@
 #define GLOBAL_PROC_REF(X) (/proc/##X)
 
 #endif
+
+#if (DM_VERSION == 515)
+/// fcopy will crash on 515 linux if given a non-existant file, instead of returning 0 like on 514 linux or 515 windows
+/// var case matches documentation for fcopy.
+/world/proc/__fcopy(Src, Dst)
+	if (!fexists(Src))
```

Comment:
fexists might be expecting a string, whereas fcopy can take file objects as src, which may be causing problems. You might want to do something like this instead:
```suggestion
	if (istext(Src) && !fexists(Src))
```

Reply:
https://github.com/Watermelon914/tgstation/actions/runs/5166381657/jobs/9306483444
Made the changes on my master and ran the integration tests and they passed. This works as a solution (probably).
I'm going to make a PR and ask keyholders to TM until you're able to make the changes on your PR